src/views/DrefApplicationExport/index.tsx

```diff
@@ -0,0 +1,977 @@
+import { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { DownloadLineIcon } from '@ifrc-go/icons';
+import {
+    isDefined,
+    isFalsyString,
+    isTruthyString,
+} from '@togglecorp/fujs';
+
+import Container from '#components/Container';
+import Image from '#components/Image';
+import Link from '#components/Link';
+import Header from '#components/Header';
+import TextOutput from '#components/TextOutput';
+import NumberOutput from '#components/NumberOutput';
+import DescriptionText from '#components/domain/DescriptionText';
+import BlockTextOutput from '#components/domain/BlockTextOutput';
+import useTranslation from '#hooks/useTranslation';
+import { useRequest } from '#utils/restRequest';
+import { DREF_TYPE_ASSESSMENT, DREF_TYPE_IMMINENT } from '#utils/constants';
+
+import ifrcLogo from '#assets/icons/ifrc-square.png';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+/*
+TODO:
+- Footer with page number and logo
+- Separate font for headings
+- Translation
+*/
+
+// eslint-disable-next-line import/prefer-default-export
+export function Component() {
+    const { drefId } = useParams<{ drefId: string }>();
+    const [previewReady, setPreviewReady] = useState(false);
+    const strings = useTranslation(i18n);
+
+    const {
+        // pending: fetchingDref,
+        response: drefResponse,
+    } = useRequest({
+        skip: isFalsyString(drefId),
+        url: '/api/v2/dref/{id}/',
+        pathVariables: isDefined(drefId) ? {
+            id: drefId,
+        } : undefined,
+        onSuccess: () => {
+            async function waitForImages() {
+                const images = document.querySelectorAll('img');
+                const promises = Array.from(images).map(
+                    (image) => (
+                        new Promise((accept) => {
+                            image.addEventListener('load', () => {
+                                accept(true);
+                            });
+                        })
+                    ),
+                );
+
+                await Promise.all(promises);
+                setPreviewReady(true);
+            }
+
+            waitForImages();
```

Comment:
This is not a safe way to wait for image load. The component can be dismounted while the images are being loaded.

Reply:
Also, what happens if there is an error when downloading images? We should use try catch to avoid crashes

Reply:
We should also see if we can use AbortController to abort these promises when the page is dismounted.